packages/react/src/ReactFetch.js

```diff
@@ -0,0 +1,130 @@
+          // Ideally we should be able to check this on the Request but
+          // it always gets initialized with its own signal so we don't
+          // know if it's supposed to override - unless we also override the
+          // Request constructor.
+          return originalFetch(resource, options);
+        }
+        // Normalize the Request
+        let url: string;
+        let cacheKey: string;
+        if (typeof resource === 'string' && !options) {
+          // Fast path.
+          cacheKey = simpleCacheKey;
+          url = resource;
+        } else {
+          // Normalize the request.
+          const request = new Request(resource, options);
+          if (
+            (request.method !== 'GET' && request.method !== 'HEAD') ||
+            // $FlowFixMe: keepalive is real
+            request.keepalive
+          ) {
+            // We currently don't dedupe requests that might have side-effects. Those
+            // have to be explicitly cached. We assume that the request doesn't have a
+            // body if it's GET or HEAD.
+            // keepalive gets treated the same as if you passed a custom cache signal.
+            return originalFetch(resource, options);
+          }
+          cacheKey = generateCacheKey(request);
+          url = request.url;
+        }
+        const cache = dispatcher.getCacheForType(createFetchCache);
+        const cacheEntries = cache.get(url);
+        let match;
+        if (cacheEntries === undefined) {
+          // We pass the original arguments here in case normalizing the Request
+          // doesn't include all the options in this environment.
+          match = originalFetch(resource, options);
+          cache.set(url, [cacheKey, match]);
+        } else {
+          // We use an array as the inner data structure since it's lighter and
+          // we typically only expect to see one or two entries here.
+          for (let i = 0, l = cacheEntries.length; i < l; i += 2) {
+            const key = cacheEntries[i];
+            const value = cacheEntries[i + 1];
+            if (key === cacheKey) {
+              match = value;
+              // I would've preferred a labelled break but lint says no.
+              return match.then(response => response.clone());
+            }
+          }
+          match = originalFetch(resource, options);
+          cacheEntries.push(cacheKey, match);
+        }
+        // We clone the response so that each time you call this you get a new read
+        // of the body so that it can be read multiple times.
+        return match.then(response => response.clone());
+      };
+    } catch (error) {
+      // Log even in production just to make sure this is seen if only prod is frozen.
+      // eslint-disable-next-line react-internal/no-production-logging
+      console.warn(
+        'React was unable to patch the fetch() function in this environment. ' +
+          'Suspensey APIs might not work correctly as a result.',
```

Comment:
Is this the first time we've used "Suspensey" in public facing messages? I think I dig it